app/models/manageiq/providers/redhat/infra_manager/disk_attachment_builder.rb

```diff
@@ -0,0 +1,48 @@
+class ManageIQ::Providers::Redhat::InfraManager::DiskAttachmentBuilder
+  def initialize(options = {})
+    @size_in_mb = options[:size_in_mb]
+    @storage = options[:storage]
+    @name = options[:name]
+    @thin_provisioned = options[:thin_provisioned]
+    @bootable = options[:bootable]
+    @active = options[:active]
+    @interface = options[:interface]
+  end
+
+  def disk_attachment
+    thin_provisioned = self.class.true?(@thin_provisioned)
+    {
+      :bootable  => self.class.true?(@bootable),
+      :interface => @interface || "VIRTIO",
+      :active    => @active,
+      :disk      => {
+        :name             => @name,
+        :provisioned_size => self.class.mb_to_gb(@size_in_mb),
+        :sparse           => thin_provisioned,
+        :format           => self.class.disk_format_for(@storage, thin_provisioned),
+        :storage_domains  => [:id => ManageIQ::Providers::Redhat::InfraManager.extract_ems_ref_id(@storage.ems_ref)]
+      }
+    }
+  end
+
+  def self.true?(obj)
+    obj.to_s == "true"
```

Comment:
present? won't work since the value might be string 'true' or boolean true or false/'false'.
I'll suggest an alternative on the next patch to comply with `@thin_provisioned.true? / @bootable.true?`